Northstar.CustomServers/mod/scripts/vscripts/mp/_classic_mp_dropship_intro.gnut

```diff
@@ -132,22 +147,44 @@ void function EndIntroWhenFinished()
 	ClassicMP_OnIntroFinished()
 }
 
+void function SpawnPlayerOnGround( entity player )
+{
+	player.EndSignal( "OnDestroy" )
+
+	if ( IsAlive( player ) )
+		player.Die()
+
+	RespawnAsPilot( player )
+	player.FreezeControlsOnServer()
+	AddCinematicFlag( player, CE_FLAG_CLASSIC_MP_SPAWNING )
+	ScreenFadeFromBlack( player, 0.5, 0.5 )
+
+	svGlobal.levelEnt.WaitSignal( "GameStateChanged" ) // wait for game start
+	player.UnfreezeControlsOnServer() // restore player movements
+	RemoveCinematicFlag( player, CE_FLAG_CLASSIC_MP_SPAWNING )
+	TryGameModeAnnouncement( player ) // do gamemode announcement
+}
+
 void function SpawnPlayerIntoDropship( entity player )
 {
 	player.EndSignal( "OnDestroy" )
 
 	if ( IsAlive( player ) )
 		player.Die() // kill them so we don't have any issues respawning them later
 	
-	player.s.dropshipIntroIsJumping <- false
-	OnThreadEnd( function() : ( player )
+	table results = {}
+	results.isJumping <- false
```

Comment:
```suggestion
	table results = {
	  isJumping = false
	}
```

might even be worth to remove the table entirely and just use a bool

Reply:
a single variable will be reset to the value before `OnThreadEnd()`
for example:
```
int a = 1
OnThreadEnt()
{
    function(): ( a )
    {
        print( a )
    }
}
a++
```
we will print a "1" instead of "2", and I found that a table can resolve this